### ISSUE TITLE
Getting started guide could be more specific.

### ISSUE THREAD

**Issue by patxipierce** [sgrif/diesel.rs-website#85]
When running `$ cargo install diesel_cli` on a Debian 9 machine I was getting:

```
= note: /usr/bin/ld: cannot find -lsqlite3
          /usr/bin/ld: cannot find -lpq
          /usr/bin/ld: cannot find -lmysqlclient
          collect2: error: ld returned 1 exit status
```

It might be useful if the guide mentioned what command to run to solve these issues, instead of "_You can resolve this issue by either installing the library (using the usual way to do this depending on your operating system)_".

In my case the issue was resolved by installing the following:

```
sudo apt install libpq-dev libmysqlclient-dev libsqlite3-dev
```

**Comment by weiznich** [comment]
In my opinion we should not do this, because we cannot list the required libs and their install commands for every operation system out there (and keep that up to date).

**Comment by patxipierce** [comment]
True, it would be impossible to keep updated. Maybe a link to the issues section, where solutions like the above can be posted by other people.

Loving diesel btw. ^_^

**Comment by pete-lennart** [comment]
I am getting started with diesel and had a lot of trouble setting up diesel_cli (getting the above mentioned errors) when installing everything in the recommended way on a conventional Windows 10 and Ubuntu 18 OS. It was really unpleasant, especially since I found the solutions to problems on both installations only by googling multiple different keywords, and I considered dumping diesel altogether multiple times (having the impression, diesel is just too immature).
Although I can understand, that there can't be solutions for problems on all available operating systems out there, the documentation should be valid for the mostly used operating systems (not just macOS) or should at least provide **some kind of help for the reader to guide him to possible solutions**. Be it some kind of FAQ or just a link to an issue section like @patxipierce suggested. Otherwise some people using Windows or Linux might be thinking diesel is just a poorly written piece of code, because, well, it can't even be compiled...

**Comment by weiznich** [comment]
In my personal opinion the error messages and the help statement are pretty clear, but I've done some c/c++ in the past, so I'm used to linker error messages. 
That said: I'm not sure how exactly we could provide more help than saying: You must install the libraries requested by the linker. If you have a concrete suggestion please open a PR. 
Putting that as platform specific hint into some kind of FAQ wont solve the original problem of incomplete and probably outdated instructions.

**Comment by ryanphillips838** [comment]
I just submitted a pull request that adds the one sentence that helped me get the cli installed. I didn't see anywhere on the getting started where it says you need openssl installed and that's where it kept failing for me. It's short and sweet and contains enough information for a user to go find the tools they need to get started even with little knowledge about linkers.

**Comment by nico-abram** [comment]
Would additional note/s on windows specific build instructions for required libraries that require manual compilation/installation be desireable too? (Such as using vcpkg)

**Comment by weiznich** [comment]
@nico-abram As already pointed out above multiple times: I'm not willing to keep a list of operating system specific instructions there as I cannot keep that up to date. I do not see what would make windows there be special.

**Comment by RKochenderfer** [comment]
I believe a simple solution is to just provide a link to https://github.com/diesel-rs/diesel/blob/master/guide_drafts/backend_installation.md would help with the issue. I created a pull request to just add a link to the documents to the diesel_cli installation instructions on the getting started page.
